test/inductor/test_auto_functionalize.py

```diff
@@ -964,6 +966,219 @@ def f(x):
+
+                v2 = torch.compile(
+                    func, backend=counter_v2, fullgraph=True, dynamic=True
+                )
+
+                inputs = [
+                    torch.rand(10, 10),
+                    torch.rand(100, 100),
+                    torch.rand(10, 2),
+                    torch.rand(1000, 1000),
+                ]
+
+                with torch._inductor.config.patch(enable_auto_functionalized_v2=True):
+                    for input in inputs:
+                        v2(input)
+
+                torch._dynamo.reset()
+
+                with torch._inductor.config.patch(enable_auto_functionalized_v2=False):
+                    for input in inputs:
+                        v1(input)
+
+                self.assertEqual(counter_v2.frame_count, counter_v1.frame_count)
+
+                self.assertEqual(counter_v1.frame_count, expected)
+
+            def func(x):
+                a = x[0]
+                b = x[1]
+                torch.ops.mylib.foo(a, b)
+
+            run_and_compare(func)
+
+            def func(x):
+                a = torch.ops.aten.alias.default(x)
+                b = torch.ops.aten.alias.default(x)
+                torch.ops.mylib.foo(a, b)
+
+            run_and_compare(func)
+
+            def func(x):
+                # last row
+                a = x[x.size()[0] - 1]
+
+                # first row
+                b = x[0]
+                torch.ops.mylib.foo(a, b)
+
+            run_and_compare(func)
+
+            def func(x):
+                a = torch.ops.aten.slice.Tensor(x, 1, 3, 4)
+                b = torch.ops.aten.slice.Tensor(x, 0, 1, 4)
+                torch.ops.mylib.foo(a, b)
+
+            # recompile here is not triggered by auto_functionalize
+            # [__recompiles]     - 0/0: 4 <= L['x'].size()[1]  # a = torch.ops.aten.slice.Tensor(x, 1, 3, 4)
+            # test/inductor/test_auto_functionalize.py:1160 in func (_decomp/decompositions.py:781 in slice_forward)
+            run_and_compare(func, 2)
+
+    # Test that the alias optimization, were alias is called instead of as_strided, preserve the fact
+    # that id(x) != id(base)
+    @torch._inductor.config.patch(enable_auto_functionalized_v2=True)
+    @unittest.skip(
+        reason="This test fails because something else in inductor optimize out the alias. issue #137434"
+    )
+    def test_alias_id(self):
+        with torch.library._scoped_library("mylib", "FRAGMENT") as lib:
+            torch.library.define(
+                "mylib::not_eq",
+                "(Tensor(a!) x, Tensor(b!) y) -> ()",
+                tags=torch.Tag.pt2_compliant_tag,
+                lib=lib,
+            )
+
+            @torch.library.impl("mylib::not_eq", "cpu", lib=lib)
+            @torch._dynamo.disable
+            def not_eq_impl(x, y):
+                self.assertNotEqual(id(x), id(y))
+
+            def func(x):
+                a = torch.ops.aten.alias.default(x)
+                torch.ops.mylib.not_eq(a, x)
+
```

Comment:
can you return `a`? and then assert that the input and output of `func` have different id identities.

Inductor doesn't match the tensor identity of intermediates of the function, but it should better match the identity of inputs/outputs of the compiled function.

Reply:
i will add another test that does that

Reply:
added that , that one pass.